### DISCUSSION TITLE
Make proofs collapsible in HTML

### DISCUSSION THREAD

**Discussion by adityam** [quarto-dev/quarto-cli#5272] [Q&A] [answered]
I am trying to port by hand maintained course website to quarto. To keep the main discussion short, I like to use `<details> <summary>` tags to keep the proofs hidden. See [this page](https://adityam.github.io/stochastic-control/inf-mdp/discounted-mdp/#prop:1) for an example. 

Is it possible to do this in quarto; either by using a custom `.scss` file or using a lua filter?

**Comment by mcanouil** [comment, marked as answer]
You could use collapsible callout to include your proof bellow or inside the theorem, see <https://quarto.org/docs/authoring/cross-references.html#theorems-and-proofs> and <https://quarto.org/docs/authoring/callouts.html>.

Here an example:

<details><summary>See the Quarto document.</summary>

````qmd
---
title: "Demo Quarto"
format: html
---

## Section

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam quis semper nisl. Nulla facilisi. Donec auctor, nisl eget aliquam tincidunt, nunc nisl aliquam nisl, eget aliquam nunc nisl eget nunc. Nu [...]


::: {#thm-line}

## Line

The equation of any straight line, called a linear equation, can be written as:

$$
y = mx + b
$$

::::{.callout-note collapse="true"}
## Proofs

Some proofs are given here.
::::

:::

See @thm-line.
````

</details>

https://user-images.githubusercontent.com/8896044/233775381-0a435cb1-8217-41ef-b59a-5e4739fc5d0a.mov

**Comment by adityam** [comment]
Thanks! Works perfectly

**Comment by ute** [comment]
I have just (three weeks late) released an extension: [custom-numbered-blocks](https://github.com/ute/custom-numbered-blocks) where you can define an environment "Proof"  in the yaml, and then use it as a class for fenced divs. It renders html as `<details><summary>...` 
You can also number theorems, propositions, etc jointly. Maybe it helps for more complicated cases :-)

![image](https://github.com/quarto-dev/quarto-cli/assets/5145859/66abe1b5-7cf5-4839-9ead-77ad29dbc8d2)

(shameless advertisement :-))
